cdc_rsync_server/cdc_rsync_server.cc

```diff
@@ -619,27 +726,25 @@ absl::Status CdcRsyncServer::HandleSendMissingFileData() {
     }
 
     status = path::StreamWriteFileContents(*fp, handler);
-    fclose(*fp);
-    if (!status.ok()) {
-      return WrapStatus(status, "Failed to write file %s", filepath);
-    }
+    finalize_pool.QueueTask(std::make_unique<FinalizeCopiedFileTask>(
+        *fp, file, filepath, is_executable, status));
+    finalize_pool.WaitForQueuedTasksAtMost(kMaxQueueSize);
 
-    // Set file write time.
-    status = path::SetFileTime(filepath, file.modified_time);
-    if (!status.ok()) {
-      return WrapStatus(status, "Failed to set file mod time for %s", filepath);
+    // Drain finalize pool for the last file.
+    if (server_index + 1 == diff_.missing_files.size()) {
+      finalize_pool.Wait();
     }
 
-    // Set executable bit, but just print warnings as it's not critical.
-    if (is_executable) {
-      path::Stats stats;
-      status = path::GetStats(filepath, &stats);
-      if (status.ok()) {
-        status = path::ChangeMode(filepath, stats.mode | kExecutableBits);
-      }
-      if (!status.ok()) {
-        LOG_WARNING("Failed to set executable bit on '%s': %s", filepath,
-                    status.ToString());
+    // Check the results of completed tasks.
+    std::unique_ptr<Task> task;
+    while ((task = finalize_pool.TryGetCompletedTask()) != nullptr) {
+      const FinalizeCopiedFileTask* finalize_task =
+          static_cast<FinalizeCopiedFileTask*>(task.get());
+      if (!finalize_task->Status().ok()) {
+        // Close and finish files that have already been copied, so we don't
+        // several already copied files because one failed.
```

Comment:
There seems to be a verb missing in the comment.

Reply:
Done.